### PR TITLE
1 hashmap

### PR DESCRIPTION
**Pull request by varchenkomichael**

### CODE REVIEW

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 1]
```diff
@@ -1,7 +1,221 @@
...
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
 
-    private final Entry[] table = new Entry[SIZE];
-}
+public class MikeHashMap<K, V> extends AbstractMap<K, V> implements Map<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+
+    private int CAPACITY;
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java]: Please, read about Java convention (code style).

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 2]
```diff
@@ -1,7 +1,221 @@
...
+
+    public int getCAPACITY() {
+        return CAPACITY;
+    }
+
+    public void setCAPACITY(int CAPACITY) {
+        this.CAPACITY = CAPACITY;
+    }
+
+    public MikeHashMap() {
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java:15]: Why did you create empty constructor?

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 3]
```diff
@@ -1,7 +1,221 @@
...
+public class MikeHashMap<K, V> extends AbstractMap<K, V> implements Map<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+
+    private int CAPACITY;
+
+    public MikeHashMap(int CAPACITY) {
+        this.CAPACITY = CAPACITY;
+    }
+
+    public int getCAPACITY() {
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java]: .

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 4]
```diff
@@ -1,7 +1,221 @@
...
+    public void setCAPACITY(int CAPACITY) {
+        this.CAPACITY = CAPACITY;
+    }
+
+    public MikeHashMap() {
+    }
+
+    private MikeEntry<K, V>[] bucket;
+
+    private MikeEntry<K, V>[] getBucket() {
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java]: Why do you need this method?

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 5]
```diff
@@ -1,7 +1,221 @@
...
+        return bucket;
+    }
+
+
+    private int indexFor(Object k) {
+        if (k != null) {
+            return k.hashCode() % DEFAULT_CAPACITY;
+        }
+        return 0;
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java:30]: It's Anti-pattern - Magic digit

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 6]
```diff
@@ -1,7 +1,221 @@
...
+    }
+
+    @Override
+    public V get(Object k) {
+        MikeEntry<K, V> mikeEntry = bucket[indexFor(k)];
+        while (mikeEntry != null) {
+            if (mikeEntry.getKey().equals(k)) {
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java:49]: What the problem of this code?

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 7]
```diff
@@ -1,7 +1,221 @@
...
+
+        for (MikeEntry<K, V> kvMikeEntry : mikeEntry) {
+            if (kvMikeEntry != null) {
+                entrySet.add(kvMikeEntry);
+            }
+        }
+        return entrySet;
+    }
+
+//    @Override
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java]: What is it?

File: src/main/java/com/varchenko/MikeHashMap.java [hunk 8]
```diff
@@ -1,7 +1,221 @@
...
+    @Override
+    public V merge(K key, V value, BiFunction<? super V, ? super V, ? extends V> remappingFunction) {
+        return Map.super.merge(key, value, remappingFunction);
+    }
+
+
+    public static class MikeEntry<K, V> implements Entry<K, V> {
+        private final K key;
+        private V value;
+        MikeEntry<K, V> next;
```
Comment (bryzhatov) [inline src/main/java/com/varchenko/MikeHashMap.java:191]: .

File: src/main/java/com/varchenko/MikeHashMap.java (hunk 1 above)
Comment (varchenkomichael) [reply src/main/java/com/varchenko/MikeHashMap.java]: done

File: src/main/java/com/varchenko/MikeHashMap.java (hunk 2 above)
Comment (varchenkomichael) [reply src/main/java/com/varchenko/MikeHashMap.java:15]: done

File: src/main/java/com/varchenko/MikeHashMap.java (hunk 4 above)
Comment (varchenkomichael) [reply src/main/java/com/varchenko/MikeHashMap.java]: done

File: src/main/java/com/varchenko/MikeHashMap.java (hunk 7 above)
Comment (varchenkomichael) [reply src/main/java/com/varchenko/MikeHashMap.java]: done

**Pull request closed without merging**